products-built-on-oxen/lokinet/guides/lokinet-on-a-soho-router.md

```diff
@@ -0,0 +1,325 @@
+
+### network topology
+
+the network toplogy of this box will be the following:
+
+* the upstream network going wherever it goes to get internet on `eth0`
+* LAN range is 10.200.0.1/24 on `wifi0`
+* Lokinet range is 10.100.0.1/16 on `lokinet0`
+
+note the interface names for `eth0` and `wifi0` will very likely be different, these are placeholder values for the purposes of this tutorial, please replace them with the appropriate values on your system.
+
+### system component layout
+
+the layout of the system has the following components:
+
+* dnsmasq handles dhcp and forwards dns from LAN to lokinet and unbound.
+* lokinet handles DNS for lokinet.
+* unbound provides encrypted dns tunnel for ICANN.
+* hostapd handles wifi access point.
+* ifdown handles managing netowrk interfaces.
+
+## setup
+
+the setup have a few steps:
+
+* installing the base system and base packages
+* installing and configuring `lokinet`
+* configuring the system to be a wifi ap with `hostapd`
+* configuring the dns with `unbound`
+* configuring the DHCP server with `dnsmasq`
+* configuring the network interface management with `ifdown`
+
+
+### install base system
+
+first install debian bookworm onto your system. you will NEED the install medium including the extra firmware [here](https://cdimage.debian.org/cdimage/unofficial/non-free/cd-including-firmware/).
+when installing you will want a minimal install with an SSH server and basic cli utils.
+
+after completing install and rebooting the system into a fresh install, update the system:
+
+`# apt update && apt full-upgrade`
+
+install ssh onto the box:
+
+`# tasksel install ssh-server`
+
+it is highly recommended (but optional) to remove any non required packages that may have been installed.
+
+`# tasksel remove desktop`
+
+now, install the required packages from debian repo for our wifi setup:
+
+`# apt install firmware-linux fwupd hostapd wireless-tools iw wpasupplicant dnsmasq unbound ntpd iptables ifdown`
+
+finally you will want to enable ip forwarding in a persistent way:
+
+`# echo 'net.ipv4.ip_forward=1' > /etc/sysctl.d/99-ip-forward.conf`
+
+on system start up that rule will be applied, but you should apply them immediately with:
+
+`# sysctl -p`
+
+
+### lokinet
+
+we now need to install and configure lokinet.
+
+#### install lokinet
+
+we add the apt repo for lokinet. follow instructions [here](https://deb.oxen.io/) to add the repo, dont install any packages from those instrctions. once you add the repos, update apt with:
+
+`# apt update`
+
+now install the lokinet package this way:
+
+`# apt install lokinet --no-install-recommends`
+
+#### set up lokinet configs
+
+the easiest way to configure lokinet is to use config override files and not touch `lokineti.ini` at all.
+that directory we try to load from if it exists is located at `/var/lib/lokinet/conf.d`. by default as of `0.9.11` tag it does not get autocreated.
+
+so first, we create the config override dir:
+
+`# mkdir -p /var/lib/lokinet/conf.d`
```

Comment:
I hate this suggestions, and we need to fix it in lokinet: config files do not belong in a program's data-dir.